pkg/dmaasbackup/periodic_backup.go

```diff
@@ -130,6 +129,19 @@ func (d *dmaasBackup) cleanupPeriodicSchedule(dbkp *v1alpha1.DMaaSBackup) error
 		return nil
 	}
 
+	if !dbkp.Spec.PeriodicFullBackupCfg.DisableSuccessfulBackupRetain {
+		successfulScheduleIdx, err := d.getSuccessfulScheduleIdx(dbkp)
+		if err != nil {
+			d.logger.Debugf("Failed to get successful backup index err=%v",
+				err)
+			return nil
+		}
+
+		if requiredSchedule <= successfulScheduleIdx {
+			requiredSchedule = successfulScheduleIdx + 1
```

Comment:
to make this indexing logic same across periodic and non-periodic, can we remove doing '+1' here, also in getting 'requriedSchedule' and 'scheduleCount', and, do '+1' at the range loop

Reply:
or do you see any issue with that?

Reply:
a log here as well similar to non-periodic case to detect one side of edge.

Reply:
ok